File: Floyd.py
#!/bin/python3

import math
import os
import random
import re
import sys
MAX_D = 100000000000007

def floyd(edges, n):
    n += 1
    for k in range(1, n):
        for i in range(1, n):
            for j in range(1, n):
                if dist[i][j] > dist[i][k] + dist[k][j]:
                    dist[i][j] = dist[i][k] + dist[k][j]

    for i in range(1, n):
        for j in range(1, n):
            if dist[i][j] == MAX_D:
                dist[i][j] = -1

if __name__ == '__main__':
    n, m = map(int, input().split())

    dist = [[MAX_D for x in range(0, n + 5)] for x in range(0, n + 5)]
    for i in range(1, n):
        dist[i][i] = 0

    for i in range(m):
        xyd = sys.stdin.readline().split()
        x = int(xyd[0])
        y = int(xyd[1])
        d = int(xyd[2])
        dist[x][y] = d

    floyd(dist, n)

    fptr = open(os.environ['OUTPUT_PATH'], 'w')
    q = int(sys.stdin.readline())

    for q_itr in range(q):
        xy = sys.stdin.readline().split()

        x = int(xy[0])

        y = int(xy[1])

        fptr.write(str(dist[x][y]))
        fptr.write('\n')
    fptr.close()

File: Floyd.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException{
        int MAX_V = 1000000009;
        String[] roadNodesEdges = scanner.nextLine().split(" ");
        int n = Integer.parseInt(roadNodesEdges[0].trim());
        int m = Integer.parseInt(roadNodesEdges[1].trim());

        int x, y, d;
        int[][] dist = new int[n + 5][n + 5];
        n++;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                dist[i][j] = MAX_V;

        for (int i = 0; i < n; i++)
            dist[i][i] = 0;

        for (int i = 0; i < m; i++) {
            String[] roadFromToWeight = scanner.nextLine().split(" ");
            x = Integer.parseInt(roadFromToWeight[0].trim());
            y = Integer.parseInt(roadFromToWeight[1].trim());
            d = Integer.parseInt(roadFromToWeight[2].trim());
            dist[x][y] = d;
        }


        for (int k = 0; k < n; k++)
            for (int i = 0; i < n; i++)
                for (int j = 0; j < n; j++)
                    if (dist[i][j] > dist[i][k] + dist[k][j])
                        dist[i][j] = dist[i][k] + dist[k][j];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                if (dist[i][j] == MAX_V)
                    dist[i][j] = -1;

        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int q = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int qItr = 0; qItr < q; qItr++) {
            String[] xy = scanner.nextLine().split(" ");

            x = Integer.parseInt(xy[0]);
            y = Integer.parseInt(xy[1]);
            bufferedWriter.write(String.valueOf(dist[x][y]));
            bufferedWriter.newLine();
        }
        
        bufferedWriter.close();
        scanner.close();
    }
}
